store/src/java/com/zimbra/cs/lmtpserver/ExternalEmailWarning.java

```diff
@@ -173,26 +186,39 @@ public String getUpdatedContent(String content) {
         if (content != null) {
             final Matcher ctTextPlainMatcher = CONTENT_TYPE_TEXT_PLAIN_PATTERN.matcher(content);
             if (ctTextPlainMatcher.find()) {
+                String textPlainHeader = ctTextPlainMatcher.group();
                 final int end = ctTextPlainMatcher.end();
-                final StringBuilder sb = new StringBuilder();
-                sb.append(content.substring(0, end));
-                if (getTextPlainWarning() != null)
-                    sb.append(getTextPlainWarning());
-                sb.append(content.substring(end));
-                content = sb.toString();
+                if (textPlainHeader!=null && textPlainHeader.contains(CONTENT_TRANSFER_BASE64_ENCODING)) {
+                    content = appendWarning(content, end, getEncodedTextPlainWarning());
+                } else {
+                    content = appendWarning(content, end, getTextPlainWarning());
+                }
             }
             final Matcher ctTextHtmlMatcher = CONTENT_TYPE_TEXT_HTML_PATTERN.matcher(content);
+            final Matcher ctTextHtmlWithoutBodyTagMatcher = CONTENT_TYPE_TEXT_HTML_WIHOUT_BODY_TAG_PATTERN.matcher(content);
             if (ctTextHtmlMatcher.find()) {
                 final int end = ctTextHtmlMatcher.end();
-                final StringBuilder sb = new StringBuilder();
-                sb.append(content.substring(0, end));
-                if (getTextHtmlWarning() != null)
-                    sb.append(getTextHtmlWarning());
-                sb.append(content.substring(end));
-                content = sb.toString();
+                content = appendWarning(content, end, getTextHtmlWarning());
+            } else if (ctTextHtmlWithoutBodyTagMatcher.find()) {
+                String textHtmlHeader = ctTextHtmlWithoutBodyTagMatcher.group();
+                final int end = ctTextHtmlWithoutBodyTagMatcher.end();
+                if (textHtmlHeader!=null && textHtmlHeader.contains(CONTENT_TRANSFER_BASE64_ENCODING)) {
```

Comment:
Validate if **Content-Transfer-Encoding:** is present but it's not equal to base64

Reply:
I validated for all encoding types i.e bit 7, bit 8, quoted printable and base64